test/subspace-test-client/src/domain_chain_spec.rs

```diff
@@ -84,9 +84,6 @@ pub fn testnet_evm_genesis() -> GenesisConfig {
         },
         ethereum: Default::default(),
         base_fee: Default::default(),
-        self_domain_id: evm_domain_test_runtime::SelfDomainIdConfig {
-            // Id of the genesis domain
-            domain_id: Some(DomainId::new(0)),
-        },
+        self_domain_id: Default::default(),
```

Comment:
Why do you even need to specify it here? It seemed like it doesn't matter what it is set to anyway.

Reply:
This field is set to `Some(DomainId::new(123))` in a later commit for testing purposes, but yeah we can use `..Default::default()` to replace `pallet_xxx: Default::default()`

Reply:
Why is this field necessary here at all? Wouldn't it be overridden during domain instantiation anyway?

Reply:
It is set to `Some(DomainId::new(123))` for testing if it will be overridden, in the production chain spec we can replace them with `..Default::default()`

Reply:
If it is overridden or doesn't matter in either case, why is it there at all?

Reply:
Sorry, I'm not following, it is a field of `RuntimeGenesisConfig` so it should be either specified with a value `pallet_domain_id: ...` or use `..Default::default()`, and it is set to `Some(DomainId::new(123))` here for testing purposes.